helix-rest/src/test/java/org/apache/helix/rest/server/TestMSDADistributedLeaderElection.java

```diff
@@ -0,0 +1,225 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import javax.ws.rs.core.Response;
+
+import org.apache.helix.msdcommon.constant.MetadataStoreRoutingConstants;
+import org.apache.helix.msdcommon.exception.InvalidRoutingDataException;
+import org.apache.helix.rest.common.ContextPropertyKeys;
+import org.apache.helix.rest.common.HelixRestNamespace;
+import org.apache.helix.rest.common.ServletType;
+import org.apache.helix.rest.server.auditlog.AuditLogger;
+import org.apache.helix.rest.server.filters.CORSFilter;
+import org.apache.helix.rest.server.resources.mock.MockMetadataStoreDirectoryAccessor;
+import org.apache.helix.zookeeper.api.client.HelixZkClient;
+import org.apache.helix.zookeeper.datamodel.ZNRecord;
+import org.apache.helix.zookeeper.datamodel.serializer.ZNRecordSerializer;
+import org.apache.helix.zookeeper.impl.factory.DedicatedZkClientFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.apache.helix.msdcommon.constant.MetadataStoreRoutingConstants.LEADER_ELECTION_ZNODE;
+
+
+public class TestMSDADistributedLeaderElection extends MetadataStoreDirectoryAccessorTestBase {
+  private final String MOCK_URL_PREFIX = "/mock";
+
+  private HelixRestServer _mockHelixRestServer;
+  private String _mockBaseUri;
+  private String _leaderBaseUri;
+  private CloseableHttpClient _httpClient;
+  private HelixZkClient _zkClient;
+
+  @BeforeClass
+  public void beforeClass() throws Exception {
+    super.beforeClass();
+    _leaderBaseUri = getBaseUri().toString();
+
+    // Start a second server for testing Distributed Leader Election for writes
+    _mockBaseUri =
+        getBaseUri().getScheme() + "://" + getBaseUri().getHost() + ":" + (getBaseUri().getPort()
+            + 1);
+    try {
+      List<HelixRestNamespace> namespaces = new ArrayList<>();
+      // Add test namespace
+      namespaces.add(new HelixRestNamespace(TEST_NAMESPACE,
+          HelixRestNamespace.HelixMetadataStoreType.ZOOKEEPER, _zkAddrTestNS, false));
+      _mockHelixRestServer =
+          new MockHelixRestServer(namespaces, getBaseUri().getPort() + 1, getBaseUri().getPath(),
```

Comment:
We see a lot of `getPort() + 1`'s here. Let's just create a new variable,

int newPort = getPort() + 1; and use newPort throughout? This will make your code more readable.

Reply:
Sure.